C/plugins/storage/sqlite/common/readings_catalogue.cpp

```diff
@@ -1933,6 +1960,52 @@ ReadingsCatalogue::tyReadingReference  ReadingsCatalogue::getReadingReference(Co
 
 }
 
+/**
+ * Get Empty Reading Table
+ *
+ * @param emptyTableReference  An empty reading table reference to be used for the given asset_code
+ * @return         True of success, false on any error
+ *
+ */
+bool ReadingsCatalogue::getEmptyReadingTableReference(tyReadingReference &emptyTableReference)
+{
+	bool isEmptyTableAvailable = false;
+	sqlite3 *dbHandle;
+	string sql_cmd;
+	sqlite3_stmt *stmt;
+
+	ConnectionManager *manager = ConnectionManager::getInstance();
+	Connection *connection = manager->allocate();
+	dbHandle = connection->getDbHandle();
+	
+	for (auto &item : m_AssetReadingCatalogue)
+	{
+		int tableId = item.second.first;
+		int dbId = item.second.second;
+		sql_cmd = "SELECT COUNT(*) FROM (SELECT 0 FROM readings_" + to_string(dbId) + ".readings_" + to_string(dbId) + "_" + to_string(tableId) + " LIMIT 1)";
```

Comment:
Why the nested select?

Reply:
asset_reading_catalogue table is just having only details to get the table not the actual table. That's why inner query it getting actual table and outer query is checking the number of rows from the table returned by inner query